showcase/tests/integration/components/hds/advanced-table/index-test.js

```diff
@@ -0,0 +1,951 @@
+    setSortableTableData(this);
+
+    await render(
+      hbs`<Hds::AdvancedTable
+  id='data-test-advanced-table'
+  @model={{this.model}}
+  @columns={{this.columns}}
+  @valign='middle'
+/>`
+    );
+
+    assert
+      .dom('#data-test-advanced-table')
+      .hasClass('hds-advanced-table--valign-middle');
+  });
+
+  test('it should render with a CSS class appropriate if no @valign value is set', async function (assert) {
+    setSortableTableData(this);
+    await render(
+      hbs`<Hds::AdvancedTable
+  id='data-test-advanced-table'
+  @model={{this.model}}
+  @columns={{this.columns}}
+/>`
+    );
+    assert
+      .dom('#data-test-advanced-table')
+      .hasClass('hds-advanced-table--valign-top');
+  });
+
+  test('it throws an assertion if @isStriped and has nested rows', async function (assert) {
+    const errorMessage =
+      '@isStriped must not be true if there are nested rows.';
+
+    setNestedTableData(this);
+    assert.expect(2);
+    setupOnerror(function (error) {
+      assert.strictEqual(error.message, `Assertion Failed: ${errorMessage}`);
+    });
+    await render(hbs`<Hds::AdvancedTable
+      id='data-test-advanced-table'
+      @isStriped={{true}}
+      @selectableColumnKey={{this.selectableColumnKey}}
+      @onSelectionChange={{this.onSelectionChange}}
+      @onSort={{this.onSort}}
+      @model={{this.model}}
+      @columns={{array (hash key='name' label='Name') (hash key='age' label='Age')}}
+    >
+      <:body as |B|>
+        <B.Tr @selectionKey={{B.data.id}} @isSelected={{B.data.isSelected}}>
+          <B.Td>{{B.data.name}}</B.Td>
+          <B.Td>{{B.data.age}}</B.Td>
+        </B.Tr>
+      </:body>
+    </Hds::AdvancedTable>`);
+
+    assert.throws(function () {
+      throw new Error(errorMessage);
+    });
+  });
+
+  test('it should support splattributes', async function (assert) {
+    setSortableTableData(this);
+    await render(
+      hbs`<Hds::AdvancedTable
+  id='data-test-advanced-table'
+  @model={{this.model}}
+  @columns={{this.columns}}
+  aria-label='data test table'
+/>`
+    );
+    assert
+      .dom('#data-test-advanced-table')
+      .hasAttribute('aria-label', 'data test table');
+  });
+
+  test('it should render with a CSS class appropriate for the @hasStickyHeader argument', async function (assert) {
+    setSortableTableData(this);
+
+    await render(
+      hbs`<div id='short-advanced-table-wrapper' style="height: 75px; overflow-y: scroll;"><Hds::AdvancedTable
+  id='data-test-advanced-table'
+  @model={{this.model}}
+  @columns={{this.columns}}
+  @hasStickyHeader={{true}}
+>
+<:body as |B|>
+    <B.Tr>
+      <B.Td>{{B.data.artist}}</B.Td>
+      <B.Td>{{B.data.album}}</B.Td>
+      <B.Td>{{B.data.year}}</B.Td>
+    </B.Tr>
+  </:body>
+</Hds::AdvancedTable></div>`
+    );
+
+    assert
+      .dom('#data-test-advanced-table .hds-advanced-table__thead')
+      .hasClass('hds-advanced-table__thead--sticky');
+
+    // commenting out for now because it is flaky
```

Comment:
Question: this test worked... fairly reliably when I ran `ember t -s`, but with just `ember t` it always fails. Should I try to debug more? or is it ok to not test that the pinned class gets added when you scroll with a sticky header.

Reply:
if you'd like, you can replace `test(` with `skip(` to further investigate a solution but not block the PR on it